src/extension/kernel.ts

```diff
@@ -596,4 +600,17 @@ export class Kernel implements Disposable {
   getSupportedLanguages() {
     return this.#controller.supportedLanguages
   }
+
+  registerTerminal(terminal: RunmeTerminal, executionId: number) {
+    const exists = this.activeTerminals.find((t) => t.executionId === executionId)
+    if (!exists) {
+      this.activeTerminals.push({ ...terminal, executionId })
```

Comment:
```suggestion
      this.activeTerminals.push(terminal)
```
One last question, is there any reason you have to destructure here? Since `executionId` hasn't changed

Reply:
but, the executionId is coming from a TaskExecution not terminal, hence we need to associate the executionId to the ActiveTerminal added to the array.